src/CodeGen/GenerateSyntax.hs

```diff
@@ -108,7 +138,7 @@ addTickIfNecessary s
 
 -- | Convert snake_case string to CamelCase Name
 toName :: String -> Name
-toName = mkName . toCamelCase
+toName = mkName . addTickIfNecessary . toCamelCase
 
 -- | Prepend "Anonymous" to named node when false, otherwise use regular toName
 toName' :: MkNamed -> String -> Name
```

Comment:
💡 We could maybe consider refactoring this into something which operates on a `MkType`, and then introduce a function of type `MkDatatype -> MkType` to deal with the uses in `datatypeForConstructors`, above. Not a blocker for this PR, just a thought.

Reply:
Will handle on separate PR.